mypy does not recognize __qualname__

1. Given Python file `test1.py`
```Python
class A(object):
    NAME = __qualname__
```
2. run mypy checking

       mypy test1.py

3. **_BUG:_** mypy reports `not defined`

       test1.py:2: error: Name '__qualname__' is not defined

The Python interpreter runs the code without error.

`__qualname__` has [been available since Python 3.3](https://docs.python.org/3/library/stdtypes.html#definition.__qualname__).

Tested on _Python 3.7.1_, using _mpypy 0.670_.

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -5569,7 +5569,7 @@ class SemanticAnalyzer(
             if not suppress_errors:
                 self.name_not_defined(name, ctx)
             return None
-        # 2. Class attributes (if within class definition)
+        # 2a. Class attributes (if within class definition)
         if self.type and not self.is_func_scope() and name in self.type.names:
             node = self.type.names[name]
             if not node.implicit:
@@ -5579,6 +5579,9 @@ class SemanticAnalyzer(
                 # Defined through self.x assignment
                 implicit_name = True
                 implicit_node = node
+        # 2b. Class attributes __qualname__ and __module__
+        if self.type and not self.is_func_scope() and name in {"__qualname__", "__module__"}:
+            return SymbolTableNode(MDEF, Var(name, self.str_type()))
         # 3. Local (function) scopes
         for table in reversed(self.locals):
             if table is not None and name in table:

--- a/test-data/unit/check-classes.test
+++ b/test-data/unit/check-classes.test
@@ -8001,3 +8001,10 @@ f5(1)  # E: Argument 1 to "f5" has incompatible type "int"; expected "Integral" 
        # N: Types from "numbers" aren't supported for static type checking \
        # N: See https://peps.python.org/pep-0484/#the-numeric-tower \
        # N: Consider using a protocol instead, such as typing.SupportsFloat
+[case testImplicitClassScopedNames]
+class C:
+    __module__
+    __qualname__
+    def f(self) -> None:
+        __module__  # E: Name "__module__" is not defined
+        __qualname__  # E: Name "__qualname__" is not defined

--- a/test-data/unit/check-classes.test
+++ b/test-data/unit/check-classes.test
@@ -8001,10 +8001,11 @@ f5(1)  # E: Argument 1 to "f5" has incompatible type "int"; expected "Integral" 
        # N: Types from "numbers" aren't supported for static type checking \
        # N: See https://peps.python.org/pep-0484/#the-numeric-tower \
        # N: Consider using a protocol instead, such as typing.SupportsFloat
+
 [case testImplicitClassScopedNames]
 class C:
-    __module__
-    __qualname__
+    reveal_type(__module__)  # N: Revealed type is "builtins.str"
+    reveal_type(__qualname__)  # N: Revealed type is "builtins.str"
     def f(self) -> None:
         __module__  # E: Name "__module__" is not defined
         __qualname__  # E: Name "__qualname__" is not defined

